django/contrib/auth/admin.py

```diff
@@ -124,7 +123,9 @@ def add_view(self, request, form_url='', extra_context=None):
     def user_change_password(self, request, id, form_url=''):
         if not self.has_change_permission(request):
             raise PermissionDenied
-        user = get_object_or_404(self.get_queryset(request), pk=id)
+        user = self.get_object(request, id)
```

Comment:
I see that most of the admin views `unquote` the object `id` before calling `get_object`.  Should we do this here?